lib/quoil/cli.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Quoil.CLI do
```

Comment:
I recognised this as the major problem, but I simply ran out of time because I needed to get some data to back my conclusions when speaking to the Telstra techs about my internet problems.

### `Ping` Module

Creating a new `Ping` module was number 1 on my TO-DO list... mainly because I wanted to add a way of implementing the ping functionality without needing to run shell commands - please see my questions on your suggestion on this matter below.

### `terminate_early?` function

I wasn't happy about cluttering up `Quoil.CLI` with the `terminate_early?` function and all the error handling either, but I was a little worried about hiding it in a module - I was sort of thinking that by having something like that as part of the chain, then it makes it clear that _this is a point where you could have early termination_... maybe I am overthinking it, but I thought that since this is a major branch point for the program, it might be helpful to somehow highlight it - **should I be even worried about this when writing more complex code?**

(I guess I could just have a comment after the call to `parse_args`, but I thought the idea was to have self-commenting code... once again, probably overthinking it, and I just need the sort of perspective that comes from years of development work ;) )